### PR TITLE
Complete API integration

### PR DESCRIPTION
**Pull request by SaurusXI**
@L04DB4L4NC3R please review

### CODE REVIEW

File: settings/api/api.go [hunk 1]
```diff
@@ -45,7 +46,7 @@ func main() {
 	ctrl := controller.New(model)
 	routes.MakeRoutes(router, ctrl)
 
-	log.Printf("Socrates API serving on PORT %s", viper.GetString("API_PORT"))
+	fmt.Printf("Socrates API serving on PORT %s", viper.GetString("API_PORT"))
```
Comment (L04DB4L4NC3R) [inline settings/api/api.go:49]: Don't use fmt. Use log.Info

File: settings/tui/screen/requests.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+func (s *Screen) putUpdate() {
+	client := &http.Client{}
+
+	body, err := json.Marshal(s.config)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	req, err := http.NewRequest(http.MethodPut, "http://localhost:" + s.port + "/update", bytes.NewBuffer(body))
```
Comment (L04DB4L4NC3R) [inline settings/tui/screen/requests.go:19]: Have there be a global config for feeding in host ans port in information. It'll add the ability to decouple the backends even more

**Review by L04DB4L4NC3R — CHANGES_REQUESTED** [review]

File: settings/api/api.go (hunk 1 above)
Comment (SaurusXI) [reply settings/api/api.go:49]: This is only here to help me debug, all the useful info is being logged. Hence the `fmt.Printf`

File: settings/tui/screen/requests.go (hunk 2 above)
Comment (SaurusXI) [reply settings/tui/screen/requests.go:19]: There is a global config being loaded in `main`. I added a port parameter to `Screen` for better readability, and passed this parameter to the constructor call in `main`.

Comment (L04DB4L4NC3R) [reply settings/tui/screen/requests.go:19]: Adding a host parameter to the viper config will ensure decoupling if someone wants it

Comment (SaurusXI) [reply settings/tui/screen/requests.go:19]: Host parameter? What would that denote?

**Pull request merged**